influxql/ast.go

```diff
@@ -137,6 +135,13 @@ type Statement interface {
 	RequiredPrivileges() ExecutionPrivileges
 }
 
+// HasDefaultDB provides an interface to get the default DB from a Statement.
+type HasDefaultDB interface {
```

Comment:
+1 for using `Database` since it's consistent with the rest of the codebase. As for the naming, if it has a `DefaultDatabase()` function then it should probably be named `DatabaseDefaulter`? That's kind of weird i suppose. Could all `Statement`s return a default database but just have it return `nil`?